src/jackdaw/admin.clj

```diff
@@ -6,11 +6,54 @@
   Like the underlying `AdminClient` API, this namespace is subject to
   change and should be considered of alpha stability."
   {:license "BSD 3-Clause License <https://github.com/FundingCircle/jackdaw/blob/master/LICENSE>"}
-  (:require [jackdaw.data :as jd])
-  (:import [org.apache.kafka.clients.admin
-            AdminClient DescribeTopicsOptions]))
-
-(set! *warn-on-reflection* true)
+  (:require
+   [jackdaw.data :as jd]
+   [manifold.deferred :as d])
+  (:import [org.apache.kafka.clients.admin AdminClient
+            DescribeTopicsOptions DescribeClusterOptions DescribeConfigsOptions]))
+
+(defprotocol Client
+  (alter-topics* [this topics])
+  (create-topics* [this topics])
+  (delete-topics* [this topics])
+  (describe-topics* [this topics])
+  (describe-configs* [this configs])
+  (describe-cluster* [this])
+  (list-topics* [this]))
+
+(def client-impl
```

Comment:
It might be nice to follow the same convention for this def.

Should it be `->` or `->>`

Should it be on one line or multiple lines?
e.g
`(-> (.alterConfigs this topics) .all deref)`
or
``` 
(-> (.describeConfigs this configs (DescribeConfigsOptions.))
                              .all .get)
```

Reply:
Agree it would be nice to get some consistency here.  I've gone for two lines. The first being the API call while the second "extracts" the value in a way that might differ for each type of command (mostly it's just `.all .deref` but in some cases the logical equivalent is a bit different like the describeCluster which returns a map of futures which are forced by the call to `datafy`).

```
(->> (.apiCall this args (Options.))
         .extract-result)
```

The idea here is to perform the bare minimum API call and extract the result without any kind of transformation